core/src/lightclient/mod.rs

```diff
@@ -354,4 +367,130 @@ impl StarknetClient for StarknetClientImpl {
+            LightClientError::OtherError(anyhow::anyhow!(
+                "Failed to convert Starknet block hash to FieldElement: {}",
+                e
+            ))
+        })?;
+        let starknet_tx_receipt = self.client.get_transaction_receipt(hash_felt).await?;
+
+        match starknet_tx_receipt {
+            MaybePendingTransactionReceipt::Receipt(receipt) => {
+                match receipt {
+                    StarknetTransactionReceipt::Invoke(InvokeTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::L1Handler(L1HandlerTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::Declare(DeclareTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::Deploy(DeployTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::DeployAccount(
+                        DeployAccountTransactionReceipt {
+                            transaction_hash,
+                            actual_fee,
+                            status,
+                            block_hash,
+                            block_number,
+                            ..
+                        },
+                    ) => {
+                        res_receipt.transaction_hash =
+                            Some(PrimitiveH256::from_slice(&transaction_hash.to_bytes_be()));
+                        res_receipt.gas_used = Some(felt_to_u256(actual_fee));
+                        res_receipt.status_code = match status {
+                            StarknetTransactionStatus::Pending => Some(U64::from(0)),
+                            StarknetTransactionStatus::AcceptedOnL1 => Some(U64::from(1)),
+                            StarknetTransactionStatus::AcceptedOnL2 => Some(U64::from(1)),
+                            StarknetTransactionStatus::Rejected => Some(U64::from(0)),
+                        };
+                        res_receipt.block_hash =
+                            Some(PrimitiveH256::from_slice(&block_hash.to_bytes_be()));
```

Comment:
Nice workaround with PrimitiveH256, could you create an issue to prepare the time where we'll unify our type handling? 

What I mean is an issue that looks like: Go from PrimitiveH256 and H256 => only one H256 type (when reth is updated probably)